src/use-swr.ts

```diff
@@ -276,14 +276,25 @@ function useSWR<Data = any, Error = any>(
     fn = config.fetcher
   }
 
+  const willRevalidateOnMount = () => {
+    return (
+      config.revalidateOnMount ||
+      (!config.initialData && config.revalidateOnMount === undefined)
+    )
+  }
+
   const resolveData = () => {
     const cachedData = cache.get(key)
     return typeof cachedData === 'undefined' ? config.initialData : cachedData
   }
 
+  const resolveIsValidating = () => {
+    return !!cache.get(keyValidating) || willRevalidateOnMount()
```

Comment:
if `config.revalidateOnMount` is `true` then initial isValidating will always be `true`? The initial revalidation is executed in effects, I'd assume there's still a `isValidating = false` in the first render?

Reply:
Yep yep I agree with you. But I'm thinking is that possible to let this feature built inside SWR?

Reply:
Yeah it avoids an extra re-render if we have this built-in :)